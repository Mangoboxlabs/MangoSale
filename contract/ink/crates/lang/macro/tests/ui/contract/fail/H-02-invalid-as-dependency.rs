use ink_lang as ink;

#[ink::contract(compile_as_dependency = "yes")]
mod invalid_as_dependency {
    #[ink(storage)]
    pub struct InvalidAsDependency {}

    impl InvalidAsDependency {
        #[ink(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn message(&self) {}
    }
}

fn main() {}
