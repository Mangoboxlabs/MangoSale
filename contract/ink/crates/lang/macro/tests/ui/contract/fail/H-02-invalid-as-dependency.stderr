error: expected a bool literal for `compile_as_dependency` ink! configuration argument
 --> $DIR/H-02-invalid-as-dependency.rs:3:17
  |
3 | #[ink::contract(compile_as_dependency = "yes")]
  |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
